execute_workflow_iteratively fails if run twice

### Version Checks (indicate both or one)

- [x] I have confirmed this bug exists on the lastest commit on the master branch

- [x] I have confirmed this bug exists on the lastest commit on the dev branch.

- [x] I have confirmed this bug exists for a specific version the software.


### Issue Description

If one runs execute_workflow_iteratively twice, we get an 
> AssertionError: placements must not have 'RESKit_sim_order' attribute

This is because the placements dataframe passed by the user is overwritten during the code execution in RESKit.

Should we automatically remove any columns called RESKit_sim_order before running the workflow?
Otherwise we could just create a copy of it within RESKit.


### Reproducible Example

```python
import reskit as rk
import pandas as pd
import numpy as np

# Create Placements DataFrame with turbine locations and specifications

placements = pd.DataFrame(
    {
        "lon": [5.985195, 5.994685, 6.004750],
        "lat": [50.797254, 50.794208, 50.784432],
        "hub_height": [120, 120, 82],
        "capacity": [4000, 4000, 4000],
        "rotor_diam": [130, 150, 136],
        "powerCurve": [np.nan, np.nan, "V117-3300"],
    }
)
from reskit.workflow_manager import execute_workflow_iteratively

workflow_args = {
    "placements": placements,
    "era5_path": rk.TEST_DATA["era5-like"],
    "gwa_100m_path": rk.TEST_DATA["gwa100-like.tif"],
    "height_scaling_data": {
        50: rk.TEST_DATA["gwa50-like.tif"],
        200: rk.TEST_DATA["gwa200-like.tif"],
    },
}

execute_workflow_iteratively(
    rk.wind.wind_era5_PenaSanchezDunkelWinklerEtAl2025,
    "era5_path",
    zoom=4,
    **workflow_args,
)
execute_workflow_iteratively(
    rk.wind.wind_era5_PenaSanchezDunkelWinklerEtAl2025,
    "era5_path",
    zoom=4,
    **workflow_args,
)
```

### Expected Behavior

No error.

### pip list -v

```bash

```

--- a/reskit/util/weather_tile.py
+++ b/reskit/util/weather_tile.py
@@ -4,6 +4,7 @@ import geokit as gk
 import osgeo
 import pandas as pd
 from smopy import deg2num
+import numpy as np
 
 
 def get_tile_XY(zoom, lon=None, lat=None, geom=None):
@@ -114,7 +115,10 @@ def get_dataframe_with_weather_tilepaths(placements, weather_path, zoom):
         assert "geom" in placements.columns or all([c in placements.columns for c in ["lon", "lat"]]), (
             f"pd.DataFrame must contain 'geom' or 'lat' and 'lon' columns."
         )
-        assert not "RESKit_sim_order" in placements.columns, f"placements must not have 'RESKit_sim_order' attribute"
+        if "RESKit_sim_order" in placements.columns:
+            assert (placements["RESKit_sim_order"].to_numpy() == np.arange(len(placements))).all(), (
+                "'RESKit_sim_order' must equal range(len(placements))"
+            )
         if not "lon" in placements.columns:
             placements["lon"] = placements.geom.apply(lambda x: x.GetX())
         if not "lat" in placements.columns:
@@ -146,16 +150,18 @@ def get_dataframe_with_weather_tilepaths(placements, weather_path, zoom):
                 axis=1,
             )
     else:
-        # make sure we have no source column to avoid overwriting data
-        assert not "source" in placements.columns, (
-            f"If weather_path is given, placements must not have a 'source' attribute already"
-        )
-
-        # add source column with the actual tile filepaths
-        placements["source"] = placements.apply(
+        # make sure we have either no source column to avoid overwriting data, or it is the correct one already
+        source = placements.apply(
             lambda x: _get_tilepath(weather_path=weather_path, zoom=zoom, lon=x.lon, lat=x.lat),
             axis=1,
         )
+        if "source" in placements.columns:
+            assert (placements["source"] == source).all(), (
+                "If weather_path is given, placements must not have a 'source' attribute already, or it needs to be the same as derived from weather_path"
+            )
+        else:
+            # add source column with the actual tile filepaths
+            placements["source"] = source
 
     # add an id column to ensure correct order preservation
     # placements["RESKit_sim_order"] = range(len(placements)) #TODO remove

